analysis/study_definition.py

```diff
@@ -1,41 +1,55 @@
 
-# Import functions
-
 from cohortextractor import (
-    StudyDefinition, 
-    patients, 
-    codelist, 
-    codelist_from_csv
+    StudyDefinition,
+    patients,
+    codelist_from_csv,
+    codelist,
+    filter_codes_by_category,
+    combine_codelists,
 )
 
-# Import codelists
+# Important Dates
+campaign_start = "2020-12-07" # change this if you need to
+latest_date = "2021-01-13" # for expectations only atm
 
-chronic_cardiac_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-cardiac-disease.csv", system="ctv3", column="CTV3ID"
-)
-chronic_liver_disease_codes = codelist_from_csv(
-    "codelists/opensafely-chronic-liver-disease.csv", system="ctv3", column="CTV3ID"
-)
-salbutamol_codes = codelist_from_csv(
-    "codelists/opensafely-asthma-inhaler-salbutamol-medication.csv",
-    system="snomed",
-    column="id",
-)
-systolic_blood_pressure_codes = codelist(["2469."], system="ctv3")
-diastolic_blood_pressure_codes = codelist(["246A."], system="ctv3")
+# Import Codelists
+from codelists import *
 
 
 # Specifiy study defeinition
-
 study = StudyDefinition(
     # Configure the expectations framework
     default_expectations={
-        "date": {"earliest": "1900-01-01", "latest": "today"},
-        "rate": "exponential_increase",
+        "date": {"earliest": "1970-01-01", "latest": latest_date},
+        "rate": "uniform",
+        "incidence": 0.2,
     },
+
     # This line defines the study population
-    population=patients.registered_with_one_practice_between(
-        "2019-02-01", "2020-02-01"
+    population=patients.satisfying(
+        """
+        registered = 1
+        AND
+        has_follow_up
+        AND
+        (age >= 18 AND age <= 110)
+        AND
+        NOT has_died
+        """
+    ),
+    has_follow_up=patients.registered_with_one_practice_between(
```

Comment:
do we want to make sure patients have follow up (or death) post vaccination? Otherwise for pts who are vaccinated and then move away we may have no data on their outcomes / VE

Reply:
What do you think? Index date plus 1 month?

Reply:
I think we'll want to look more than a month post vaccine date for infections (eg. many people will be waiting 3 mo for second dose after which their protection should improve).. I think we need "registered at latest available date" but rather than being built into inclusion criteria this could be left as an additional field people can choose to use for VE but not use for vaccine uptake studies?

Reply:
We removed `has_follow_up` from population criteria and placed this as a variable within the study defintion, but not a population exclusion

Reply:
also added in a `registered_at_latest_date` variable into the main study definition.